src/Weft/Generics/JSONResponse.hs

```diff
@@ -0,0 +1,58 @@
+module Weft.Generics.JSONResponse (
+  HasJSONResponse,
+  jsonResponse
+) where
+
+import Weft.Internal.Types
+import           GHC.Generics
+import           Data.Aeson
+import Data.Typeable
+import qualified Data.Text as T
+import qualified Data.Map as M
+
+
+------------------------------------------------------------------------------
+-- |
+type HasJSONResponse record = (Generic (record 'Response), GResToJSON (Rep (record 'Response)))
+
+jsonResponse :: (HasJSONResponse record) => (record 'Response) -> Value
+jsonResponse = gResToJSON . from
+
+class GResToJSON (ri :: * -> *) where
+    gResToJSON :: ri x -> Value
+
+instance GResToJSON ri
+      => GResToJSON (M1 a b ri) where
+  gResToJSON (M1 r) = gResToJSON r
+
+instance (GResToJSON ri, GResToJSON rj)
+      => GResToJSON (ri :*: rj) where
+  gResToJSON (ri :*: rj) = (gResToJSON ri) `combine` (gResToJSON rj)
+
+combine :: Value -> Value -> Value
+combine (Object a) (Object b) = Object $ a <> b
+combine _ (Object b)          = Object b 
+combine (Object a) _          = Object a
+combine _ _                   = Object mempty
```

Comment:
let's put an error here; it should never happen and if it does we're going to be very confused as to why